app/services/data_exports.rb

```diff
@@ -0,0 +1,293 @@
+require 'fileutils'
+require 'securerandom'
+require 'csv'
+
+module DataExports
+  class NoStoredFilesFoundError < StandardError; end
+
+  class DataStorage
+    # Public: downloads all transcription files for a given project,
+    # workflow, group or transcription
+    def resource_files_zip(resource)
+      # to do: include user id in top level directory name
+      directory_path = File.expand_path("~/data_exports_temp/downloaded_files/user_id_#{SecureRandom.uuid}")
+      FileUtils.mkdir_p(directory_path)
+
+      if resource.is_a?(Transcription)
```

Comment:
I think this currently has 4 different modes of behaviour depending on what you pass in. 

This method can be refactored / split out for reach resource type as you know what kind of resource at the call site, so call a different method. The common behaviour can be extracted and reused to avoid this general purpose method.